packages/flutter/test/material/dialog_test.dart

```diff
@@ -332,4 +334,47 @@ void main() {
       new Rect.fromLTRB(40.0, 24.0, 800.0 - 40.0, 600.0 - 24.0),
     );
   });
+
+  testWidgets('Dialog widget contains route semantics', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+    await tester.pumpWidget(
+      new MaterialApp(
+        theme: new ThemeData(brightness: Brightness.dark),
+        home: new Material(
+          child: new Builder(
+            builder: (BuildContext context) {
+              return new Center(
+                child: new RaisedButton(
+                  child: const Text('X'),
+                  onPressed: () {
+                    showDialog<void>(
+                      context: context,
+                      builder: (BuildContext context) {
+                        return const AlertDialog(
+                          title: const Text('Title'),
+                          content: const Text('Y'),
+                          actions: const <Widget>[ ],
+                          semanticName: 'alert',
+                        );
+                      },
+                    );
+                  },
+                ),
+              );
+            },
+          ),
+        ),
+      ),
+    );
+
+    expect(semantics, isNot(includesNodeWith(flags: <SemanticsFlag>[SemanticsFlag.isRoute], value: 'alert')));
```

Comment:
nit: I also find these slightly easier to read if the properties 'flags', and 'value' get their own line.

Reply:
done